Added changes from Spaceship upstream! 🚀⭐

Snuggle opened the pull request:
## Description
This adds a config option to enable/disable verbose versioning,
which should be disabled by default.

**Upstream pull request:**
https://github.com/denysdovhan/spaceship-prompt/pull/521

<!--- Provide a general summary of your changes in the Title above -->
## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->
The original rust section displays the verbose version by default. For example: `v1.30.0-nightly`, `v1.30.0-beta` or `v1.30.0`, depending on what branch you were using. (Stable not having a suffix)

This change makes non-verbose version numbers the default `v1.30.0`, and a config option to enable it only if wanted.

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
(All three)

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, tests ran to see how -->
<!--- your change affects other areas of the code, etc. -->
- [ ] I have tested using **MacOS**
- [x] I have tested using **Linux**

I have tested using the `./tests/run.fish` locally, as well as running logical snippets in my shell to make sure the syntax is correct.

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] I have checked that no other PR duplicates mine
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have updated the tests accordingly.

Snuggle committed:
Added changes from Spaceship upstream! 🚀⭐

This adds a config option to enable/disable verbose versioning,
which should be disabled by default.

Upstream pull request:
https://github.com/denysdovhan/spaceship-prompt/pull/521

matchai commented on functions/__sf_section_rust.fish:
```diff
@@ -32,6 +33,10 @@ function __sf_section_rust -d "Display the current Rust version"
 
 	set -l rust_version (rustc --version | cut -d' ' -f2)
```
Thinking it over, I think the "fish way" of doing this would be.

```fish
set -l rust_version (rustc --version | string split ' ')[2]
```

It's effectively the same, but would be more approachable for those familiar with fish commands. Thoughts? 🤔

matchai commented on functions/__sf_section_rust.fish:
```diff
@@ -32,6 +33,10 @@ function __sf_section_rust -d "Display the current Rust version"
 	set -l rust_version (rustc --version | cut -d' ' -f2)
 
+	if test $SPACEFISH_RUST_VERBOSE_VERSION = false
+        set rust_version (echo $rust_version | cut -d'-' -f1) # Cut off -suffixes from version. "v1.30.0-beta.11" or "v1.30.0-nightly"
```
Following that same thought process, this would then be:

```fish
set rust_version (string split '-' $rust_version)[1]

```

Snuggle commented:
OH! I totally forgot that fish already has a way to do this. Great thinking, @matchai!

Snuggle committed:
Think like a fish!

This replaces the cut command with Fish's "string" builtin.
If you have fish installed, view "file:///usr/share/doc/fish/commands.html#string"
in your browser of choice.

Co-authored-by: Matan Kushner <hello@matchai.me>

matchai reviewed (approved):
Looks good to me! 👍
